### PR TITLE
ovl_Item_Shield OK

### PR DESCRIPTION
**Pull request by Emill**

### CODE REVIEW

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 1]
```diff
@@ -49,21 +37,206 @@ static ColliderCylinderInit D_80B871A0 = {
...
+    (ActorFunc)ItemShield_Destroy,
+    (ActorFunc)ItemShield_Update,
+    (ActorFunc)ItemShield_Draw,
+};
+
+s32 unused = 0xFFFF00FF;
+s32 unused2 = 0xFF0000FF;
+
+void func_80B86F68(ItemShield* this, GlobalContext* globalCtx);
+void func_80B86BC8(ItemShield* this, GlobalContext* globalCtx);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:57]: These should go under the other prototypes, separated by a newline.

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 2]
```diff
@@ -49,21 +37,206 @@ static ColliderCylinderInit D_80B871A0 = {
...
+            for (i = 0; i < 8; i++) {
+                this->unk_19E[i] = 1 + 2 * i;
+                this->unk_1A8[i].x = Rand_CenteredFloat(10.0f);
+                this->unk_1A8[i].y = Rand_CenteredFloat(10.0f);
+                this->unk_1A8[i].z = Rand_CenteredFloat(10.0f);
+            }
+            break;
+    }
+
+    Actor_SetScale(&this->actor, 0.01f);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:91]: Either use all `thisx` in the function (if required), or replace them all with `this->actor` etc, don't mix and match.

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 3]
```diff
@@ -49,21 +37,206 @@ static ColliderCylinderInit D_80B871A0 = {
...
+void func_80B86BC8(ItemShield* this, GlobalContext* globalCtx);
+
+void func_80B86920(ItemShield* this, ItemShieldActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void ItemShield_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ItemShield* this = THIS;
+
+    s32 i;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:66]:
```suggestion
    ItemShield* this = THIS;
    s32 i;
```

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 4]
```diff
@@ -49,21 +37,206 @@ static ColliderCylinderInit D_80B871A0 = {
...
+                Actor_Kill(&this->actor);
+            }
+        }
+    }
+}
+
+void func_80B86BC8(ItemShield* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        Actor_Kill(&this->actor);
+    } else {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:129]: See above.

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 5]
```diff
@@ -49,21 +37,206 @@ static ColliderCylinderInit D_80B871A0 = {
...
+    ItemShield* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80B86AC8(ItemShield* this, GlobalContext* globalCtx) {
+    Actor_MoveForward(&this->actor);
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        Actor_Kill(&this->actor);
+    } else {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:107]: We usually do early returns after `Actor_Kill`s.

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 6]
```diff
@@ -49,21 +37,206 @@ static ColliderCylinderInit D_80B871A0 = {
...
+    }
+}
+
+void func_80B86CA8(ItemShield* this, GlobalContext* globalCtx) {
+    static Vec3f D_80B871F4 = { 0.0f, 0.0f, 0.0f };
+    static f32 D_80B87200[] = { 0.3f, 0.6f, 0.9f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f,
+                                1.0f, 0.85f, 0.7f, 0.55f, 0.4f, 0.25f, 0.1f, 0.0f };
+    static f32 D_80B87240[] = { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.8f,
+                                0.6f, 0.4f, 0.2f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f };
+
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:151]:
```suggestion
```

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 7]
```diff
@@ -49,21 +37,206 @@ static ColliderCylinderInit D_80B871A0 = {
...
+    s32 i;
+    s32 temp;
+
+    Actor_MoveForward(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 10.0f, 10.0f, 0.0f, 5);
+    this->actor.shape.yOffset =
+        ((Math_SinS(this->actor.shape.rot.x) >= 0)
+            ? Math_SinS(this->actor.shape.rot.x)
+            : -Math_SinS(this->actor.shape.rot.x))
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:160]: Can this be `ABS`?

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 8]
```diff
@@ -49,21 +37,206 @@ static ColliderCylinderInit D_80B871A0 = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Shield/func_80B86CA8.s")
+        func_80093D18(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Shield/func_80B86F68.s")
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_item_shield.c", 460),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Shield/ItemShield_Update.s")
+        gSPDisplayList(POLY_OPA_DISP++, SEGMENTED_TO_VIRTUAL(D_060224F8));
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:238]: Don't need the newlines here.

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 9]
```diff
@@ -49,21 +37,206 @@ static ColliderCylinderInit D_80B871A0 = {
...
+    OBJECT_LINK_CHILD,
+    sizeof(ItemShield),
+    (ActorFunc)ItemShield_Init,
+    (ActorFunc)ItemShield_Destroy,
+    (ActorFunc)ItemShield_Update,
+    (ActorFunc)ItemShield_Draw,
+};
+
+s32 unused = 0xFFFF00FF;
+s32 unused2 = 0xFF0000FF;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:54]: These look like RGBA8s, although since they're unused it doesn't really matter.

**Review by EllipticEllipsis — COMMENTED** [review]
Nice job with this actor. A few stylistic things.

**Comment by Emill** [comment]
Thanks for the quick review. I've now addressed the changes.

**Review by EllipticEllipsis — APPROVED** [review]

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 10]
```diff
@@ -49,21 +40,192 @@ static ColliderCylinderInit D_80B871A0 = {
...
+                this->unk_1A8[i].y = Rand_CenteredFloat(10.0f);
+                this->unk_1A8[i].z = Rand_CenteredFloat(10.0f);
+            }
+            break;
+    }
+
+    Actor_SetScale(&this->actor, 0.01f);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    osSyncPrintf(VT_SGR("32") "Item_Shild %d \n" VT_RST, this->actor.params);
```
Comment (fig02) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:93]: this vt macro should be a color. vt_fmt.py in tools will take the commands as a string
this one is `VT_FGCOL(GREEN)`

File: src/overlays/actors/ovl_Item_Shield/z_item_shield.c [hunk 11]
```diff
@@ -49,21 +40,192 @@ static ColliderCylinderInit D_80B871A0 = {
...
+    (ActorFunc)ItemShield_Update,
+    (ActorFunc)ItemShield_Draw,
+};
+
+static Color_RGBA8 unused = { 255, 255, 0, 255 };
+static Color_RGBA8 unused2 = { 255, 0, 0, 255 };
+
+void func_80B86920(ItemShield* this, ItemShieldActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
```
Comment (fig02) [inline src/overlays/actors/ovl_Item_Shield/z_item_shield.c:61]: the standard name for this func is `<Actor>_SetupAction, if you wanted to name this one now

**Comment by Emill** [comment]
Fixed!

**Review by fig02 — APPROVED** [review]

**Review by Roman971 — APPROVED** [review]

**Pull request merged**